### PR TITLE
API

### PR DESCRIPTION
**Pull request by IA-Fuad**
API to retrieve data from database

### CODE REVIEW

File: BlazorAppAPI/Services/IProjectService.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        {
+            db.Projects.Add(projectEntity);
+
+            try
+            {
+                db.SaveChanges();
+            }
+            catch (Exception)
+            {
+                throw;
```
Comment (sahariar-tomal) [inline BlazorAppAPI/Services/IProjectService.cs:55]: We should not throw exception like this. We need to handle this properly. We can use logging mechanism for this. Or we can return null object and check in controller if it is null then return response with appropriate wordings like Could not create project.

File: BlazorAppAPI/Services/IProjectService.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+            db.Entry(projectEntity).State = EntityState.Modified;
+
+            try
+            {
+                db.SaveChanges();
+            }
+            catch (Exception)
+            {
+                throw;
```
Comment (sahariar-tomal) [inline BlazorAppAPI/Services/IProjectService.cs:106]: same here.

File: BlazorAppAPI/Services/IProjectService.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+            db.Projects.Remove(project);
+
+            try
+            {
+                db.SaveChanges();
+            }
+            catch (Exception)
+            {
+                throw;
```
Comment (sahariar-tomal) [inline BlazorAppAPI/Services/IProjectService.cs:122]: Same here

**Review by sahariar-tomal — CHANGES_REQUESTED** [review]
Please resolve this issues.

File: BlazorAppAPI/Controllers/AccountController.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ public AccountController(IAccountService accountService, IConverter converter)
             this.converter = converter;
         }
 
-        [Authorize(Roles = "Admin")]
+       // [Authorize(Roles = "Admin")]
```
[the remark this answers is not in the document]
Comment (IA-Fuad) [reply BlazorAppAPI/Controllers/AccountController.cs:24]: Needed to create users for testing purposes. Authorize tag requires to pass jwt token every time to do that. I will uncomment it before merging to the master branch.

File: BlazorAppAPI/Controllers/ProjectController.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+using DataModels.EntityModels;
+using DataModels.ReponseModels;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+
+namespace BlazorAppAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProjectController : ControllerBase
```
[the remark this answers is not in the document]
Comment (IA-Fuad) [reply BlazorAppAPI/Controllers/ProjectController.cs:11]: Only the register method will be restricted to the admin. But the login method can be used by everyone.

File: BlazorAppAPI/Services/IProjectService.cs (hunk 1 above)
Comment (IA-Fuad) [reply BlazorAppAPI/Services/IProjectService.cs:55]: Working on this now.

File: BlazorAppAPI/Controllers/ProjectController.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (sahariar-tomal) [reply BlazorAppAPI/Controllers/ProjectController.cs:11]: But here we are in Projects controller and it has endpoints like GetAllProjects, GetProjectById, CreateProject, UpdateProject etc. So these should not be accessed by unauthorized users.

File: BlazorAppAPI/Controllers/AccountController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (sahariar-tomal) [reply BlazorAppAPI/Controllers/AccountController.cs:24]: You can have some local changes for faster testing but when we create PR we should not have changes like that. You can apply your local changes whenever you want.

**Review by sahariar-tomal — CHANGES_REQUESTED** [review]

File: BlazorAppAPI/Controllers/ProjectController.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (IA-Fuad) [reply BlazorAppAPI/Controllers/ProjectController.cs:11]: My bad. I thought it was AccountController. Fixing it now.

File: BlazorAppAPI/Services/ICUD.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Repo.DbContext;
+using System;
+
+namespace BlazorAppAPI.Services
+{
+    public interface ICUD
```
Comment (sahariar-tomal) [inline BlazorAppAPI/Services/ICUD.cs:7]: The name should be more meaningful.

**Review by sahariar-tomal — CHANGES_REQUESTED** [review]
The idea for a generic service for doing CRUD is really awesome. But the name should be more meaningful.

File: BlazorAppAPI/Services/ICUD.cs (hunk 6 above)
Comment (IA-Fuad) [reply BlazorAppAPI/Services/ICUD.cs:7]: Will change that.

**Review by sahariar-tomal — APPROVED** [review]
👍

**Review by sahariar-tomal — APPROVED** [review]
👍

**Pull request merged**